hazelcast-client/src/main/java/com/hazelcast/client/proxy/ClientMapProxy.java

```diff
@@ -815,13 +821,14 @@ public void loadAll(Set<K> keys, boolean replaceExistingValues) {
         MapQueryRequest request = new MapQueryRequest(name, predicate, IterationType.ENTRY);
         QueryResultSet result = invoke(request);
```

Comment:
QueryResultSet has entries in a Set which are then added to a List here. It might be another optimisation to store QueryResultSet.entries as a List when it is deserialized.